apps/documents/tasks.py

```diff
@@ -146,3 +157,116 @@ def create_collection_from_assistant_task(collection_id: int, assistant_id: int)
             CollectionFile.objects.filter(file__in=file_without_remote_ids).values_list("id", flat=True)
         )
         index_collection_files_task(collection_file_ids=file_ids_to_index)
+
+
+@shared_task(ignore_result=True)
+def sync_document_source_task(document_source_id: int):
+    """Sync a specific document source"""
+    from apps.documents.document_source_service import sync_document_source
+
+    try:
+        document_source = DocumentSource.objects.get(id=document_source_id)
+    except DocumentSource.DoesNotExist:
```

Comment:
I'm wondering if we shouldn't let it fail here if the document source doesn't exist. Reason being that `sync_document_source_task` is called either manualy or when one is created. In both scenarios, we expect this to exist, so if it doesn't there's something wrong that we should probably know about. Thougths?

Reply:
Nit: Pre-load the collection, since it's needed in the service later on

Reply:
> hs?

If we got an error from this in Sentry I don't know what we would do. Likely ignore it or track it down to some race condition. I think it's safe to ignore.

Reply:
That's fair